addons/pos_mollie/static/src/app/payment_mollie.js

```diff
@@ -0,0 +1,176 @@
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._make_mollie_payment(cid);
+    }
+
+    /**
+     * @override
+     *
+     * At the moment, Mollie POS payments are no cancellable from the API call.
+     * It can be only cancelled from the terminal itself. If you cancel the
+     * transaction from the terminal, we get webhook call for status update and
+     * `handleMollieStatusResponse` will handle cancellation.
+     *
+     * But in case you want to cancel from odoo pos (e.g terminal switched off during transaction)
+     * We show the dialog to guide user in correct direction.
+     */
+    async send_payment_cancel(order, cid) {
+        const { confirmed } = await this.env.services.popup.add(ConfirmPopup, {
+            title: _t('Cancel mollie payment'),
+            body: _t('First cancel transaction on POS device. Only use force cancel if that fails'),
+            confirmText: _t('Force Cancel'),
+            cancelText: _t('Discard')
+        });
+
+        if (confirmed) {
+            super.send_payment_cancel(order, cid);
+            const paymentLine = this.get_pending_mollie_line();
+            paymentLine.set_payment_status('retry');
+            return true;
+        }
+    }
+
+    /**
+     * Call odoo backend to create payment request on mollie.
+     */
+    _make_mollie_payment(cid) {
+        let order = this.pos.get_order();
+        if (order.selected_paymentline.amount < 0) {
+            this._show_error(_t("Cannot process transactions with negative amount."));
+            return Promise.resolve();
+        }
+
+        let payment_data = this._prepare_payment_data(cid);
+        return this.env.services.orm.silent
+            .call('pos.payment.method', 'mollie_payment_request', [
+                [this.payment_method.id],
+                payment_data
+            ])
+            .then((response_data) => {return this._mollie_handle_response(response_data);})
+            .catch(this._handle_odoo_connection_failure.bind(this));
+    }
+
+    /**
+     * Call odoo backend to create payment request on mollie.
+     */
+    _prepare_payment_data(cid) {
+        const order = this.pos.get_order();
+        const line = order.paymentlines.find((paymentLine) => paymentLine.cid === cid);
+        return {
+            'description': order.name,
+            'pos_reference': order.uid,
+            'currency': this.pos.currency.name,
+            'amount': line.amount,
+            'session_id': this.pos.pos_session.id,
+            'config_id': this.pos.pos_session.config_id[0],
+        }
+    }
+
+    _handle_odoo_connection_failure(data = {}) {
+        const line = this.get_pending_mollie_line();
+        if (line) {
+            line.set_payment_status("retry");
+        }
+        this._show_error(
+            _t("Could not connect to the Odoo server, please check your internet connection and try again.")
+        );
+        return Promise.reject(data);
+    }
```

Comment:
this is the same function as in `pos_adyen`. Can you find a way to not duplicate it? Perhaps lift it to the `PaymentInterface` class

Reply:
This method is available in several other 3 pos payment modules, i think better do it in another task.

What do you think ?

Reply:
You are right. We can merge like this and we'll handle that later